js/game.js

```diff
@@ -3,10 +3,68 @@ import Tamagotchi from "./modules/tamagotchi.js";
 export default class Game {
   constructor() {
     this.tamagotchi = new Tamagotchi();
+    this.healthIntervalId = null;
+    this.hungerIntervalId = null;
+    this.energyIntervalId = null;
+    this.funIntervalId = null;
   }
 
   start = ({ healthElement, hungerElement, energyElement, funElement }) => {
-    this.tamagotchi.mount({ healthElement });
+    this.tamagotchi.mount({
+      healthElement,
+      hungerElement,
+      energyElement,
+      funElement,
+    });
     console.log("Game started");
+
+    const decreaseHealth = () => {
+      if (this.tamagotchi.health.value <= 0) {
+        this.end({
+          healthInterval: this.healthIntervalId,
+          hungerInterval: this.hungerIntervalId,
+          energyInterval: this.energyIntervalId,
+          funInterval: this.funIntervalId,
+        });
+      } else {
+        if (
+          this.tamagotchi.hunger.value <= 0 ||
+          this.tamagotchi.energy.value <= 0
+        )
+          this.tamagotchi.health.value--;
+        this.tamagotchi.displayHealth(healthElement);
+      }
+    };
+
+    const decreaseHunger = () => {
+      if (this.tamagotchi.hunger.value > 0) this.tamagotchi.hunger.value--;
+      this.tamagotchi.displayHunger(hungerElement);
+    };
+
+    const decreaseEnergy = () => {
+      if (this.tamagotchi.energy.value > 0) {
+        this.tamagotchi.energy.value--;
+        if (this.tamagotchi.fun.value <= 0) this.tamagotchi.energy.value--;
+      }
+      this.tamagotchi.displayEnergy(energyElement);
+    };
+
+    const decreaseFun = () => {
+      if (this.tamagotchi.fun.value > 0) this.tamagotchi.fun.value--;
+      this.tamagotchi.displayFun(funElement);
+    };
+
+    this.healthIntervalId = setInterval(decreaseHealth, 1000);
+    this.hungerIntervalId = setInterval(decreaseHunger, 1000);
+    this.energyIntervalId = setInterval(decreaseEnergy, 2000);
+    this.funId = setInterval(decreaseFun, 1000);
+  };
+
+  end = ({ healthInterval, hungerInterval, energyInterval, funInterval }) => {
```

Comment:
**nitpick:** You don't have to pass intervals as arguments, because you have already an access to it by using `this` keyword. Simply speaking `this` gives you an access to things that you set in the class constructor. So it will work if you do:

```
  end = () => {
    clearInterval(this.healthInterval);
    clearInterval(this.hungerInterval);
    clearInterval(this.energyInterval);
    clearInterval(this.funInterval);
    console.log("Game over");
  };
```